src/router_core/connections.c

```diff
@@ -273,7 +289,7 @@ int qdr_connection_process(qdr_connection_t *conn)
     int event_count = 0;
 
     if (conn->closed) {
-        core->conn_close_handler(core->user_context, conn, conn->error);
+        conn->protocol_adaptor->conn_close_handler(conn->protocol_adaptor->user_context, conn, conn->error);
```

Comment:
consider caching the value of conn->protocol_adaptor in a local variable - shorter lines and maybe faster?